woocommerce/woocommerce-ios
# Dynamic Dashboard: Dashboard view in SwiftUI

Part of #12411 


## Description

This PR migrates part of the dashboard view controller into a new SwiftUI view in preparation for the dynamic dashboard project. The existing `DashboardViewController` is massive and messy so hopefully this refactoring makes updating the UI much more easily.

Due to the large size of changes, this first PR handles half of the logic on dashboard view:
- Added a new enum for dashboard cards.
- Created a SwiftUI view and hosting controller.
- Switched to the new SwiftUI view for the dashboard screen with a feature flag.
- Added the following features to the new SwiftUI view:
  - Store name subtitle: the old view hides the subtitle when the title is small. I keep the subtitle in this new screen for simplicity.
  - Tab bar item on the new view.
  - Share store button and functionality.
  - Added store stats and top performers section by wrapping the existing view controller as-is.
  - Added store onboarding card.
  - Added Blaze section.
  - Handled pull-to-refresh.

The rest of the features will be added in a subsequent PR.

## Testing instructions

- Ensure that the feature flag `dynamicDashboard` is enabled.
- Build the app and log in to a store.
- Confirm that the dashboard screen contains a correct title, subtitle, functioning share button, with 3 sections: store stats & top performers, onboarding card and Blaze section if available.
- Confirm that the pull-to-refresh feature works as before.

Disable the feature flag `dynamicDashboard` and build the app. Confirm that the dashboard screen still works as before.

## Screenshots


<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/a07fd497-f94c-4589-b6fb-aca998749487" width=320 />

---
